src/components/FriendRequests/FriendRequests.tsx

```diff
@@ -32,8 +32,14 @@ const FriendRequests :FC<FriendRequestsProps> =({incomingFriendRequests})=>{
         await apiPost(senderId, 'deny')
     }
 
+    useEffect(()=> {
+        const client = new PusherClientHandler(sessionId, requests)
+        console.log('useEffect called')
+        client.subscribeToPusherClient(setRequests)
+    }, [sessionId, requests]);
+
     return <div aria-label='friend requests'>
-        {incomingFriendRequests.length == 0 ?
```

Comment:
This was the bug that sunk some time, can this component be decomposed a bit to be cleaner?